kuka_rsi_hw_interface/src/rsi_command_handler.cpp

```diff
@@ -0,0 +1,325 @@
+  }
+  buffer_it++;
+
+  // Could not find all parameter, or found too much
+  if (isBaseLessNode && numOfParam != element.GetParams().size()) {
+    std::sprintf(
+      err_buf_,
+      "%lu parameter found. It does not match with %s elements parameter list size.", numOfParam,
+      element.GetName().c_str());
+    throw std::logic_error(err_buf_);
+  }
+
+  // Node should have childs, but did not found any
+  if (isBaseLessNode && element.GetChilds().size() > 0) {
+    std::sprintf(
+      err_buf_, "%s node should have chields but did not found any",
+      element.GetName().c_str());
+    throw std::logic_error(err_buf_);
+  }
+  // fast forward if
+  for (; *buffer_it == ' '; buffer_it++) {
+  }
+  if ((size_t)(buffer_it - buffer) >= buffer_size) {
+    throw std::range_error("Out of the buffers range");
+  }
+
+  if (!isBaseLessNode) {
+    if (*buffer_it != '<') {
+      // Node base is data
+      if (!element.CastParam(node_name, buffer_it)) {
+        auto i = std::sprintf(err_buf_, "Could not cast the ");
+        std::snprintf(err_buf_ + i, node_name.length_ + 1, "%s", node_name.data_ptr_);
+        i += node_name.length_;
+        std::sprintf(
+          err_buf_ + i, " parameter into the %s elements parameter list",
+          element.GetName().c_str());
+        throw std::logic_error(err_buf_);
+      }
+      if (*buffer_it != '<') {
+        auto i = sprintf(err_buf_, "In \"");
+        std::snprintf(err_buf_ + i, node_name.length_ + 1, "%s", node_name.data_ptr_);
+        i += node_name.length_;
+        std::sprintf(err_buf_ + i, "\" param syntax error found");
+        throw std::logic_error(err_buf_);
+      }
+    } else {
+      // node base has childs
+      for (auto && child : element.GetChilds()) {
+        decodeNode(child, buffer, buffer_it, buffer_size);
+      }
+    }
+    for (; (size_t)(buffer_it - buffer) < buffer_size && *buffer_it != '<'; buffer_it++) {
+    }
+    if ((size_t)(buffer_it - buffer) >= buffer_size) {
+      throw std::range_error("Out of the buffers range");
+    }
+    buffer_it++;
+    if (*(buffer_it) != '/') {
+      std::sprintf(
+        err_buf_,
+        "Syntax error, while node end in the %s node",
+        element.GetName().c_str());
+      throw std::logic_error(err_buf_);
+    }
+    buffer_it++;
+    if (!element.IsNameValid(node_name, buffer_it)) {
+      auto i = std::sprintf(err_buf_, "The detected name (");
+      std::snprintf(err_buf_ + i, node_name.length_ + 1, "%s", node_name.data_ptr_);
+      i += node_name.length_;
+      std::sprintf(
+        err_buf_ + i, ") does not match the elements name: %s",
+        element.GetName().c_str());
+      throw std::logic_error(err_buf_);
+    }
+    if (*buffer_it != '>') {
+      std::sprintf(
+        err_buf_,
+        "Syntax error, while node end in the %s node",
+        element.GetName().c_str());
+      throw std::logic_error(err_buf_);
+    }
+  }
+}
+
+void RSICommandHandler::encodeNode(
```

Comment:
rename to encodeNodes

Reply:
Still rename it even if the function only encode one node, and calls itself?